data-sets-api-server/src/main/java/org/zowe/data/sets/services/zosmf/PutDataSetRenameZosmfRequestRunner.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * This program and the accompanying materials are made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Copyright IBM Corporation 2019
+ */
+package org.zowe.data.sets.services.zosmf;
+
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+
+import org.apache.http.HttpStatus;
+import org.apache.http.client.methods.RequestBuilder;
+import org.apache.http.entity.ContentType;
+import org.apache.http.entity.StringEntity;
+import org.zowe.api.common.connectors.zosmf.ZosmfConnector;
+import org.zowe.api.common.errors.ApiError;
+import org.zowe.api.common.exceptions.ZoweApiErrorException;
+import org.zowe.api.common.exceptions.ZoweApiRestException;
+import org.zowe.api.common.utils.ResponseCache;
+import org.zowe.data.sets.model.DataSetRenameRequest;
+import org.zowe.data.sets.model.ZosmfRenameRequest;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+
+public class PutDataSetRenameZosmfRequestRunner extends AbstractZosmfDataSetsRequestRunner<String> {
+    private DataSetRenameRequest request;
+    private String oldDataSetName;
+
+    public PutDataSetRenameZosmfRequestRunner(String oldDataSetName, DataSetRenameRequest request) {
+        this.oldDataSetName = oldDataSetName;
+        this.request = request;
+    }
+
+    @Override
+    protected RequestBuilder prepareQuery(ZosmfConnector zosmfConnector) throws URISyntaxException, IOException {
+        String urlPath = String.format("restfiles/ds/%s", request.getNewName());
+        URI requestUrl = zosmfConnector.getFullUrl(urlPath); // $NON-NLS-1$
+        
+        JsonParser parser = new JsonParser();
+        JsonObject requestBody = (JsonObject)parser. parse(convertIntoZosmfRequestJson(oldDataSetName));
+        
+        StringEntity requestEntity = new StringEntity(requestBody.toString(), ContentType.APPLICATION_JSON);
+        RequestBuilder requestBuilder = RequestBuilder.put(requestUrl).setEntity(requestEntity);
+        
+        return requestBuilder;
+    }
+
+    @Override
+    protected int[] getSuccessStatus() {
+        return new int[] { HttpStatus.SC_OK, HttpStatus.SC_CREATED, HttpStatus.SC_NO_CONTENT };
+    }
+
+    @Override
+    protected String getResult(ResponseCache responseCache) throws IOException {
+        String response = responseCache.getEntity();
+        return response;
+    }
+
+    @Override
+    protected ZoweApiRestException createGeneralException(ResponseCache responseCache, URI uri) throws IOException {
+        return super.createGeneralException(responseCache, uri);
+    }

```

Comment:
This method isn't needed if you are just overriding to call super

Reply:
removed